src/utils/ThreadLogger.cpp

```diff
@@ -106,3 +106,13 @@ void ThreadLogger::logMessage()
     }
 }
 
+void ThreadLogger::logPerformance(const std::string& category, const std::string& data) {
+    std::ostringstream ss;
```

Comment:
Use the main log function to log messages. this is not threadsafe (hence the whole point of the logger), and using multiple methods to queue messages would not be too intuitive. You could opt to use a helper method instead